src/components/organism/SearchForm.js

```diff
@@ -0,0 +1,42 @@
+import React, { useState, useEffect } from "react";
+import Search from "../atom/Search";
+import Dropdown from "../atom/Dropdown";
+
+const SearchForm = ({ products, setFilteredProducts }) => {
+
+    const [searchInput, setSearchInput] = useState("");
+    const [productType, setProductType] = useState("");
+
+
+    useEffect(() => {
+        const filtered = products.filter((product) => {
```

Comment:
The original code here was on the right track, but just didn't function correctly. match should mainly be used for regex purposes as it either returns null, or an array of matching strings.

I switched to using .includes, fixed the issues of upper case miss matches by making everything lowercase, then filtered on a set of statements checking if search input or productType are selected, then filtering accordingly if they are. If they are not, we skip that filter alltogether. This makes sure we can do a combination of filtering on type and search, not just one or the other.

Reply:
also it increases readability.